internal/controllers/cluster/cluster_controller.go

```diff
@@ -215,6 +219,22 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster)
 func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Cluster) (reconcile.Result, error) {
 	log := ctrl.LoggerFrom(ctx)
 
+	// Call the BeforeClusterDelete hook to before proceeding further.
+	if feature.Gates.Enabled(feature.RuntimeSDK) {
```

Comment:
If we keep it. Do we want to check if the ClusterClass feature gate is enabled and if the cluster is topology managed?

I know that we want to implement it in this controller, but independent of that do we want to provide that hook to all Clusters or only ClusterClass-managed clusters?

Reply:
(cc @fabriziopandini @killianmuldoon @chrischdi)

Reply:
IMO it's a good idea to keep this to Topology managed clusters to begin with. We can remove that restriction later on if we have a good reason, but I think exposing this single hook without a good overall non-Topology hook story isn't a great UX.

Reply:
Agreed, would be better to limit it to managed clusters. Will update it.